elasticsearch/sandboxes/heka/output/elasticsearch_bulk_api.lua

```diff
@@ -144,6 +148,37 @@ end
 
 batch_count = 0
 retry       = false
+local try_count = 0
+
+local function send_batch2()
+    try_count = try_count + 1
+    local ret, err = send_batch()
+    if ret < 0 then client = nil end -- always use a new connection after an error
+
+    if ret == 0 then
+        retry = false
+    elseif discard and try_count > maxretry then
+        retry = false
+        err = "Discarding "..batch_count.." messages after "..try_count.." attemps, ret="..ret.." err="..err
+        ret = -1
+    elseif abort and try_count > maxretry then
+        retry = true -- not saved by preserve_data for now
+        error("Abort sending "..batch_count.." messages after "..try_count.." attemps, ret="..ret.." err="..err)
```

Comment:
Now that it terminates it needs proper cleanup so the next restart won't send a double size batch (clear out the batch or re-establish the batch_count on restart after termination)